static/src/javascripts/projects/commercial/modules/messenger/scroll.spec.js

```diff
@@ -0,0 +1,187 @@
+                    observe: () => {},
+                    unobserve: () => {},
+                    disconnect: () => {
+                        onIntersect = null;
+                    },
+                });
+            }
+        }
+
+        beforeEach(() => {
+            Object.defineProperty(global, 'IntersectionObserver', {
+                value: IntersectionObserver,
+                writable: true,
+            });
+            reset(true);
+            // $FlowFixMe
+            addScrollListener(iframe1, respond1);
+            // $FlowFixMe
+            addScrollListener(iframe2, respond2);
+        });
+
+        afterEach(() => {
+            Object.defineProperty(global, 'IntersectionObserver', {
+                value: null,
+                writable: true,
+            });
+        });
+
+        it('should call respond1 but not respond2 at the top of the page', done => {
+            console.log(onIntersect);
+            if (onIntersect) {
+                onIntersect([
+                    { target: iframe1, intersectionRatio: 0.5 },
+                    { target: iframe2, intersectionRatio: 0 },
+                ]);
+            }
+            onScroll()
+                .then(() => {
+                    expect(respond1).toHaveBeenCalledTimes(2);
+                    expect(respond2).toHaveBeenCalledTimes(1);
+                })
+                .then(done)
+                .catch(done.fail);
+        });
+
+        it('should call respond2 but not respond1 at the bottom of the page', done => {
+            if (onIntersect) {
+                onIntersect([
+                    { target: iframe1, intersectionRatio: 0 },
+                    { target: iframe2, intersectionRatio: 0.5 },
+                ]);
+            }
+            onScroll()
+                .then(() => {
+                    expect(respond1).toHaveBeenCalledTimes(1);
+                    expect(respond2).toHaveBeenCalledTimes(2);
+                })
+                .then(done)
+                .catch(done.fail);
+        });
+    });
+
+    describe('Without IntersectionObserver', () => {
+        beforeEach(() => {
+            reset(false);
+            // $FlowFixMe
+            addScrollListener(iframe1, respond1);
+            // $FlowFixMe
+            addScrollListener(iframe2, respond2);
+        });
+
+        it('should call respond1 but not respond2 at the top of the page', done => {
+            detect.getViewport.mockReturnValue({ width: 400, height: 300 });
+            onScroll()
+                .then(() => {
+                    expect(respond1).toHaveBeenCalledTimes(2);
+                    expect(respond2).toHaveBeenCalledTimes(1);
+                })
+                .then(done)
+                .catch(done.fail);
+        });
+
+        it('should call respond2 but not respond1 at the bottom of the page', done => {
+            detect.getViewport.mockReturnValue({ width: 400, height: 6528 });
```

Comment:
Literally mimicking the behaviour of the old tests. This test passes even when commenting out the following line (`window.scrollTo`) because the `isIframeInViewport` method only looks at the height, so something is up here.